app/controllers/application_controller/ci_processing.rb

```diff
@@ -978,6 +970,33 @@ def process_elements(elements, klass, task, display_name = nil, order_field = ni
 
   private ############################
 
+  def process_element_destroy(element, klass, name)
+    return unless element.respond_to?(:destroy)
+
+    audit = {:event        => "#{klass.name.downcase}_record_delete",
+             :message      => "[#{name}] Record deleted",
+             :target_id    => element.id,
+             :target_class => klass.base_class.name,
+             :userid       => session[:userid]}
+
+    model_name  = ui_lookup(:model => klass.name) # Lookup friendly model name in dictionary
+    record_name = get_record_display_name(element)
+
+    element.destroy
+
+    if element.destroyed?
+      AuditEvent.success(audit)
+      add_flash(_("%{model} \"%{name}\": Delete successful") % {:model => model_name, :name => record_name})
+    else
+      error_msg = ""
+      element.errors.each do |_attr, msg|
+        error_msg = msg
+      end
```

Comment:
@GregP @dclarizio are we assuming always having a single error message?  should we be defensive and just join the  error messages with some sort of separator.  Maybe something like the following
```ruby
  error_msg = element.errors.collect { |_attr, msg| msg }.join(';')
```